libc/unistd_getopt.c

```diff
@@ -314,6 +318,107 @@ TEST(unistd_getopt, getopt_unknownopt_noarg)
 }
 
 
+TEST(unistd_getopt, getopt_unknownopt_multiple)
+{
+	/* mocks of main() arguments */
+	char *testargv1[] = { "cmd", "-xxx", NULL };
+	char *testargv2[] = { "cmd", "-ab", "-c", "value", NULL };
+	int testargc1 = 2, testargc2 = 4;
+	struct getopt_t ret = { 0 };
+
+	testmain(testargc1, testargv1, &ret, "abc:");
+	TEST_ASSERT_EQUAL_INT(UNKNOWNOPT, ret.err);
+	TEST_ASSERT_EQUAL_INT(0, ret.aflag);
+	TEST_ASSERT_EQUAL_INT(0, ret.bflag);
+	TEST_ASSERT_NULL(ret.cvalue);
+	TEST_ASSERT_EQUAL_INT(0, ret.nonopts);
+
+	/* Assert that parsing argument works correctly after passing multiple unknown options
+	   There was issue related to this case */
+
+	/* default presets of results structure */
+	ret.aflag = 0;
```

Comment:
no need to reset fields individually, You're doing `memset()` 2 lines below anyway